docs/specifications/tests/Delegation-TP/delegation01.md

```diff
@@ -1,48 +1,98 @@
 
-### Inputs
+The RFC ([RFC 1034]) pre-dates IPv6. Sinces IPv4 and IPv6 work as separate networks, this
+test case has been extended to test for for two name servers that resolv into IPv4 addresses
+and IPv6 addresses respectively.
 
-The domain name to be tested.
 
-### Ordered description of steps to be taken to execute the test case
-
-1. Obtain the complete set of name servers from parent using
-   [Method2](../Methods.md).
-2. If the number of name servers (distinct NS records) returned in previous step is less than two, the
-   record as an error.
-3. Obtain the complete set of name servers from the child 
-   using [Method3](../Methods.md).
-4. If the number of name servers (distinct NS records) returned in the previous step is less than two, the
-   record as an error.
-5. If an error was recorded in step 2 or step 4, then the test case fails.
- 
-### Outcome(s)
+## Inputs
 
-If the total amount of name servers used for the domain are two or more,
-then the test case succeeds.
+The domain name to be tested.
 
-### Special procedural requirements
+## Ordered description of steps to be taken to execute the test case
+
+1. Obtain the complete set of name servers (distinct NS records) from 
+   parent using [Method2](../Methods.md).
+2. If the number of name servers from parent is at least two emit the
+   message *ENOUGH_NS_PARENT* else emit  
+   *NOT_ENOUGH_NS_PARENT*.
+3. Obtain the IP addresses for the set of the name servers from parent 
+   using [Method4](../Methods.md).
+4. If the number of name servers from parent resolving to an IPv4 
+   address is none then emit  *NO_IPV4_NS_PARENT*, else 
+   if it is one emit *NOT_ENOUGH_IPV4_NS_PARENT* else emit
+   *ENOUGH_IPV4_NS_PARENT*.
+5. If the number of name servers from parent resolving to an IPv6 
+   address is none then emit  *NO_IPV6_NS_PARENT*, else 
+   if it is one emit *NOT_ENOUGH_IPV6_NS_PARENT* else emit
+   *ENOUGH_IPV6_NS_PARENT*.
+6. Obtain the complete set of name servers from the child using 
+   [Method3](../Methods.md).
+7. If the number of name servers from child is at least two emit the
+   message *ENOUGH_NS_CHILD* else emit  
+   *NOT_ENOUGH_NS_CHILD*.
+8. Obtain the IP addresses for the set of the name servers from child 
+   using [Method5](../Methods.md) and, for out-of-bailiwick NS, using
+   [Method4](../Methods.md).
+9. If the number of name servers from child resolving to an IPv4 
+   address is none then emit  *NO_IPV4_NS_CHILD*, else 
+   if it is one emit *NOT_ENOUGH_IPV4_NS_CHILD* else emit
+   *ENOUGH_IPV4_NS_CHILD*.
+9. If the number of name servers from child resolving to an IPv6 
+   address is none then emit  *NO_IPV6_NS_CHILD*, else 
+   if it is one emit *NOT_ENOUGH_IPV6_NS_CHILD* else emit
+   *ENOUGH_IPV6_NS_CHILD*.
+
+ 
+## Outcome(s)
+
+The outcome of the test case depends on the highest level of the messages 
+generated.
+
+Message                       | Default level (if message is emitted)
```

Comment:
Oh, and it's always the default level, right? Whether or not the message is emitted.